From 4fd045536296d90c5cd7534bf210e63c8d03b370 Mon Sep 17 00:00:00 2001
From: Anderson Torres <anderson.torres.8519@gmail.com>
Date: Wed, 16 Sep 2026 19:00:25 -0300
Subject: [PATCH] Revert "le-js.el: Add support for JavaScript via Indium"

This reverts commit 6f46e116c57d7be772c1b7bc81785c01352acc0d.

Indium was frozen at 2021-03-09.  Instead of packaging Indium and its
dependencies, let's just get rid of it.

---
 le-js.el | 41 -----------------------------------------
 lispy.el |  2 --
 2 files changed, 43 deletions(-)
 delete mode 100644 le-js.el

diff --git a/le-js.el b/le-js.el
deleted file mode 100644
index 41a5984..0000000
--- a/le-js.el
+++ /dev/null
@@ -1,41 +0,0 @@
-(require 'indium)
-
-(defun lispy--eval-js (str)
-  (let ((r nil))
-    (indium-eval
-     str (lambda (value) (setq r (indium-render-remote-object-to-string value))))
-    (while (not r)
-      (accept-process-output))
-    (substring-no-properties r)))
-
-(defun lispy--eval-js-str ()
-  (if (region-active-p)
-      (lispy--string-dwim)
-    (lispy--string-dwim
-     (lispy-bounds-python-block))))
-
-(defun lispy--js-completion-at-point ()
-  (let* ((prefix (buffer-substring-no-properties
-                  (let ((bol (point-at-bol))
-                        (prev-delimiter (1+ (save-excursion
-                                              (re-search-backward "[([:space:]]" nil t)))))
-                    (if prev-delimiter
-                        (max bol prev-delimiter)
-                      bol))
-                  (point)))
-         (expression (if (string-match-p "\\." prefix)
-                         (replace-regexp-in-string "\\.[^\\.]*$" "" prefix)
-                       "this"))
-         (cands nil))
-    (indium-client-get-completion
-     expression
-     indium-debugger-current-frame
-     (lambda (candidates)
-       (setq cands candidates)))
-    (while (null cands)
-      (accept-process-output))
-    (list (- (point) (length (company-grab-symbol)))
-          (point)
-          (mapcar #'identity cands))))
-
-(provide 'le-js)
diff --git a/lispy.el b/lispy.el
index 26e96c6..b592cd5 100644
--- a/lispy.el
+++ b/lispy.el
@@ -4232,8 +4232,6 @@ SYMBOL is a string."
      le-clojure lispy-eval-clojure)
     (python-mode
      le-python lispy--eval-python lispy-eval-python-str lispy-eval-python-bnd)
-    (js2-mode
-     le-js lispy--eval-js lispy--eval-js-str)
     (julia-mode
      le-julia lispy-eval-julia lispy-eval-julia-str)
     (racket-mode
-- 
2.54.0

